ReactiveCocoa/Swift/SignalProducer.swift

```
@@ -275,16 +285,29 @@ private struct BufferState<Value, Error: ErrorType> {
 	// producer was terminated.
 	var observers: Bag<Signal<Value, Error>.Observer>? = Bag()
 
-	// Appends a new value to the buffer, trimming it down to the given capacity
-	// if necessary.
-	mutating func addValue(value: Value, upToCapacity capacity: Int) {
-		values.append(value)
-
-		let overflow = values.count - capacity
-		if overflow > 0 {
-			values.removeRange(0..<overflow)
-		}
-	}
+    mutating func addValue(value: Value) {
+        if self.capacity == 1 {
+            self.values = [value]
+            return
+        }
+        if values.count == 0 {
+            values = Array<Value>(count: self.capacity * 2, repeatedValue: value)
```

Passing `Int.max` as the `capacity` easily let this crash.